Add portmap+proxy conformance test

joestringer opened the pull request:
Extend the k8s conformance tests to include proxy + portmap tests (which currently fail due to #12700).

Related: #12714

joestringer committed:
test: Add portmap+proxy conformance test

Extend the conformance tests to add the newly introduced test which
validates proxy + portmap paths (Currently broken). Skipped via XIt for
now, remove the X when resolved.

Signed-off-by: Joe Stringer <joe@cilium.io>

christarazi commented on examples/kubernetes/connectivity-check/connectivity-check-proxy.yaml:
```diff
@@ -879,6 +879,8 @@ spec:
 kind: Service
 
+---
+
```
Seems like there's some extraneous YAML here. Is it possible to avoid these?

(Multiple instances of this throughout)

joestringer replied:
Yeah, I think the `echo-c` gets defined twice here too. Probably just need to rework the way the echo-c is generated / integrated into these broader YAMLs.

christarazi reviewed (approved):
LGTM; one minor comment

joestringer committed:
connectivity-check: Add echo-c to quarantine YAML

The quarantine YAML was missing echo-c, which meant that it would fail
to deploy properly. Fix it up.

Signed-off-by: Joe Stringer <joe@cilium.io>
